Update CONTRIBUTING.MD (#605)
Azure/sap-hana: CONTRIBUTING.md

  * detecting regressions introduced by code changes before merging them.

## Execution-focused
* We don't just work on some grand plan that may or never be completely executed;
  * It is encouraged to make changes to gradually improve old codebase to meet standards. However, we are not planning a refactor.
  * we start building out the solution and iterate towards a grand plan.

## Coding Guidelines
This repository integrates with [Azure Pipelines](https://azure.microsoft.com/en-us/services/devops/pipelines/), which invokes build checks on the submitted pull requests aginst defined branch(eg. master). :exclamation: All pull requests are required to pass the Azure pipelines test before it can be merged.

### PR Basics
This section captures fundamentals on how new features should be developed and fixes made to the codebase.

1. **Close on design before sending PRs**
	- Add and describe design by creating an issue [here](https://github.com/Azure/sap-hana/issues). Discussions on the design will happen in the issue page. How? please take a look at the [example](https://github.com/Azure/sap-hana/issues/337).

> 	- For sizable features, please request a design meeting.

Reviewer: how? is there a standard process?

Author: Updating to use sap-hana@microsoft.com

> 	- In the design meeting, the expectation is all attendees have reviewed the design. 

Reviewer: I assume the expectation is that all attendees have reviewed the design before the meeting?
```suggestion
	- In the design meeting, the expectation is all attendees have reviewed the design prior to the meeting. 
```

1. **Design for modularity, easy versioning, easy deployment and rollback**
	- The design has to make sure it is independent and has minimum impact on other modules.

> 	- There should be a set of test cases in places to prove the design works and will not break existing code.

Reviewer: ```suggestion
	- There should be a set of test cases in place to prove the design works and will not break existing code.
```

**PR guidelines**
1. Required information in PR ([example](https://github.com/Azure/sap-hana/pull/480)):
    	- Always link to the issue that is is trying to resolve with tag **Closes**.
	- Describe the **Problem** that it tries to resolve.
	- Provide the **Solution** that this PR contains.
	- Provide **Tests** that have been done to make sure this PR does not break existing code (either in master or branch). If the test requires certain instructions, please add that information as well.
1. The PRs should be easily tested independent of other projects in progress.
1. Submit PRs with small commits with descriptive comemnts, so that make it easier to rollback in case of problem.

> 1. If it requires several PRs for a design, please create a feature branch, and submit PR against the feature branch instead of master (:exclamation: *Make sure you alert the admin of the github repo to add the feature branch into Azure pipeline so we make sure your PR will be tested*).

Reviewer: how to contact the admin of the github repo?

Author: Updating to use sap-hana@microsoft.com

### PR reviews guidelines
We need to ensure quality along with agility. We need to move to everyone agreeing on the base requirement and then relying on systems in place to catch and mitigate issues.
1. Focus on the [PR Basics](#pr-basics). PRs have to adhere to Basics with no exceptions.

> 1. In additional to Basics, PR reviews need to focus on the quality of a PR. eg. caching potential issues/bugs, semantic problem, nitpick, etc...

Reviewer: ```suggestion
1. In additional to Basics, PR reviews need to focus on the quality of a PR. eg. catching potential issues/bugs, semantic problems, nitpicks, etc...
```

1.  Keep PRs in an open published state for at least one working day, which would allow everyone in other regions to review.

> 1. For hotfixes, keep PRs open for at least 4 hrs.

Reviewer: clarify that these are business hours (i.e. nights and weekends during PDT don't count)

Author: Done!
